csharp/src/Apache.Arrow/Ipc/ArrowFileReaderImplementation.cs

```diff
@@ -70,36 +72,79 @@ protected override async Task ReadSchemaAsync()
             {
                 BaseStream.Position = BaseStream.Length - ArrowFileConstants.Magic.Length - 4;
 
-                bytesRead = await BaseStream.ReadAsync(buffer, 0, 4).ConfigureAwait(false);
-                footerLength = BinaryPrimitives.ReadInt32LittleEndian(buffer);
+                bytesRead = await BaseStream.ReadAsync(buffer.AsMemory(0, 4)).ConfigureAwait(false);
+                footerLength = ReadFooterLength(buffer, bytesRead);
+            }).ConfigureAwait(false);
 
-                if (bytesRead != 4) throw new InvalidDataException(
-                    $"Failed to read footer length. Read <{bytesRead}>, expected 4.");
+            await Buffers.RentReturnAsync(footerLength, async (buffer) =>
+            {
+                _footerStartPostion = (int)BaseStream.Length - footerLength - ArrowFileConstants.Magic.Length - 4;
```

Comment:
Is buffer.Length == footerLength?  If so, you could use that instead of footerLength to avoid closing over footerLength.  If instead buffer is a byte[] rented from the pool and thus could have a length > footerLength, you might change the signature to pass in a `Memory<byte>` rather than a `byte[]`, with it sliced to the right length, and then you could use `buffer.Length` for that purpose.